File: src/python/ulisp.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
Main program to run micro-lisp interpreter, which in turn
will run automated tests for Chronian.
"""

from util import fatal_error
from reader import Reader


def main():
  """ Do everything.
  """
  reader = Reader()
  lex = reader.read_lexeme()
  while lex:
    print("< %s, %s >" % (str(type(lex)), str(lex)))
    lex = reader.read_lexeme()
  fatal_error("Done!")


main()

File: src/javascript/ulisp.js
"use strict"

var reader_module = require("./reader")
var util = require("./util")

function main() {
  var reader = new reader_module.Reader()
  var lex = reader.read_lexeme()
  while (lex) {
    console.log(lex)
    lex = reader.read_lexeme()
  }
  
  console.log("Done!")
}

main()

